templates.go

```diff
@@ -1,1606 +1,411 @@
+        --node-ip=${DEFAULT_IPV4} \
+        --api-servers=https://{{.Cluster.Kubernetes.API.Domain}} \
+        --containerized \
+        --enable-server \
+        --logtostderr=true \
+        --machine-id-file=/rootfs/etc/machine-id \
+        --cadvisor-port=4194 \
+        --healthz-bind-address=${DEFAULT_IPV4} \
+        --healthz-port=10248 \
+        --cluster-dns={{.Cluster.Kubernetes.DNS.IP}} \
+        --cluster-domain={{.Cluster.Kubernetes.Domain}} \
+        --network-plugin=cni \
+        --register-node=true \
+        --register-schedulable=false \
+        --allow-privileged=true \
+        --kubeconfig=/etc/kubernetes/config/kubelet-kubeconfig.yml \
+        --node-labels="role=master,kubernetes.io/hostname=${HOSTNAME},ip=${DEFAULT_IPV4},{{.Cluster.Kubernetes.Kubelet.Labels}}" \
+        --v=2"
+        ExecStop=-/usr/bin/docker stop -t 10 $NAME
+        ExecStopPost=-/usr/bin/docker rm -f $NAME
+        [Install]
+        WantedBy=multi-user.target
+    - name: kubeadm-init.service
+      enable: true
+      contents: |
+        [Unit]
+        Description=Initialize master with kubeadm
+        Wants=kubelet.service
+        [Service]
+        Type=oneshot
+	Environment=IMAGE=maru/kubeadm
+        Environment=NAME=%p.service
+        EnvironmentFile=/etc/network-environment
+        ExecStartPre=-/usr/bin/docker stop $NAME
+        ExecStartPre=-/usr/bin/docker rm $NAME
+        ExecStartPre=-/usr/bin/docker pull $IMAGE
+        ExecStart=/usr/bin/docker run \
+            --name $NAME \
+            $IMAGE \
+            kubeadm init --token {{.Token}}
+        [Install]
+        WantedBy=multi-user.target
+    - name: wait-for-k8s-nodes.service
+      enable: true
+      contents: |
+        [Unit]
+        Description=Wait for nodes availability
+        Wants=kubeadm-init.service
+        [Service]
+        Type=oneshot
+        Environment=IMAGE={{.Cluster.Kubernetes.Kubectl.Docker.Image}}
+	Environment=NAME=%p.service
+	EnvironmentFile=/etc/network-environment
+	ExecStartPre=-/usr/bin/docker stop $NAME
+        ExecStartPre=-/usr/bin/docker pull $NAME
+        ExecStart=/usr/bin/docker run \
+            --name $NAME \
+            $IMAGE \
+            kubectl get nodes
+        [Install]
+        WantedBy=multi-user.target
+    - name: calico-setup.service
+      enable: true
+      contents: |
+        [Unit]
+        Description=Setup self-hosted Calico
+        Wants=wait-for-k8s-nodes.service
+        [Service]
+        Type=oneshot
+        Environment=IMAGE={{.Cluster.Kubernetes.Kubectl.Docker.Image}}
+	Environment=NAME=%p.service
+	EnvironmentFile=/etc/network-environment
+	ExecStartPre=-/usr/bin/docker stop $NAME
+        ExecStartPre=-/usr/bin/docker pull $NAME
+        ExecStart=/usr/bin/docker run \
+            --name $NAME \
+            $IMAGE \
+            kubectl apply -f http://docs.projectcalico.org/v2.3/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml
```

Comment:
Is there any way to keep the manifest in the repo (or another repo), but not inside this file? can we import or do we need to host the file ourselves (which I'd also be ok with)?

I'd like to keep as much out of this as possible, so we can have a much more modular setup and upgrades (to e.g. calico) can be super easy)